src/internal/client.ts

```diff
@@ -379,4 +392,348 @@ export class TypedClient {
       }
     }
   }
+
+  private logStream?: stream.Writable
+
+  /**
+   * log the request, response, error
+   */
+  private logHTTP(reqOptions: IRequest, response: http.IncomingMessage | null, err?: unknown) {
+    // if no logStream available return.
+    if (!this.logStream) {
+      return
+    }
+    if (!isObject(reqOptions)) {
+      throw new TypeError('reqOptions should be of type "object"')
+    }
+    if (response && !isReadableStream(response)) {
+      throw new TypeError('response should be of type "Stream"')
+    }
+    if (err && !(err instanceof Error)) {
+      throw new TypeError('err should be of type "Error"')
+    }
+    const logStream = this.logStream
+    const logHeaders = (headers: RequestHeaders) => {
+      Object.entries(headers).forEach(([k, v]) => {
+        if (k == 'authorization') {
```

Comment:
should `k` be lower cased?

Reply:
this header is set by client in low case